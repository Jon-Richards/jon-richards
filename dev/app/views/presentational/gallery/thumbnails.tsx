import * as React from 'react';
import { Thumbnail, ThumbnailProps } from './thumbnail';
import { v4 as uuid } from 'uuid';

export function Thumbnails(props: ThumbnailsProps): JSX.Element {
  const mapped = props.data.map(thumbnail => {
    const { sources, fallbackSource, altText, href } = thumbnail;

    return (
      <Thumbnail
        key={uuid()}
        sources={sources}
        fallbackSource={fallbackSource}
        altText={altText}
        href={href}
        onClick={e => props.clickHandler(e, href)}
      />
    );
  });

  return <React.Fragment>{mapped}</React.Fragment>;
}

export type ThumbnailsProps = {
  data: ThumbnailShape[],
  clickHandler: (
    e: React.MouseEvent<HTMLElement, MouseEvent>,
    href: string
  ) => {}
}

type ThumbnailShape = Omit<ThumbnailProps[][0], 'onClick'>;
