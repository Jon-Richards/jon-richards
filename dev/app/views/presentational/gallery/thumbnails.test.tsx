import * as React from 'react';
import { render } from 'enzyme';
import { Thumbnails } from './thumbnails';

describe('The Thumbnails component.', () => {
  it('Should match the snapshot.', () => {
    const wrapper = render(
      <Thumbnails
        data={[
          {
            title: 'A thumbnail',
            sources: [
              { mediaQuery: '(min-width: 300)', source: './image_3.jpg' },
              { mediaQuery: '(min-width: 200)', source: './image_2.jpg' },
              { mediaQuery: '(min-width: 100)', source: './image_1.jpg' },
            ],
            fallbackSource: './bogus.jpg',
            description: 'A brief description.',
            href: 'https://www.google.com'
          }
        ]}
        clickHandler={() => ({})}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render one thumbnail per element in the data prop.', () => {
    const wrapper = render(
      <div>
        <Thumbnails
          data={[
            {
              title: 'A thumbnail',
              sources: [
                { mediaQuery: '(min-width: 300)', source: './image_3.jpg' },
                { mediaQuery: '(min-width: 200)', source: './image_2.jpg' },
                { mediaQuery: '(min-width: 100)', source: './image_1.jpg' },
              ],
              fallbackSource: './bogus.jpg',
              description: 'A brief description.',
              href: 'https://www.google.com'
            },
            {
              title: 'Another thumbnail',
              sources: [
                { mediaQuery: '(min-width: 300)', source: './image_3.jpg' },
                { mediaQuery: '(min-width: 200)', source: './image_2.jpg' },
                { mediaQuery: '(min-width: 100)', source: './image_1.jpg' },
              ],
              fallbackSource: './bogus.jpg',
              description: 'A brief description.',
              href: 'https://www.google.com'
            }
          ]}
          clickHandler={() => ({})}
        />
      </div>
    );

    const thumbs = wrapper.find('[data-testid="gallery__thumb"]');
    expect(thumbs.length).toBe(2);
  });
});
