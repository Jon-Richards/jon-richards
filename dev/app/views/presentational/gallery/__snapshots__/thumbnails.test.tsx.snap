// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Thumbnails component. Should match the snapshot. 1`] = `
.emotion-8 {
  display: inline-block;
  width: 100%;
  height: auto;
  padding: 0rem;
  margin: 0rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 100%;
  height: auto;
  position: relative;
  padding: 1px;
  margin: 0rem;
  overflow: hidden;
  background-color: hsla(0,0%,14%,1);
  background-image: linear-gradient( 170deg, hsla(0,0%,30%,1), hsla(0,0%,10%,1), hsla(0,0%,17%,1) );
  box-shadow: 0.4rem 0.8rem 2rem 0rem hsla(0,0%,0%,0.5);
}

.emotion-6 {
  display: block;
  width: 100%;
  height: 100%;
  margin: 0rem;
  padding: 0rem;
  background-color: hsla(0,0%,14%,1);
  background-image: linear-gradient( 170deg, hsla(0,0%,14%,1), hsla(0,0%,8%,1) );
}

.emotion-5 {
  position: relative;
}

.emotion-1 {
  width: 100%;
  height: auto;
  padding: 0rem;
  margin: 0rem;
}

.emotion-0 {
  display: block;
  width: 100%;
  height: auto;
}

.emotion-4 {
  position: absolute;
  top: 0rem;
  left: 0rem;
  width: 100%;
  height: 100%;
  padding: 2rem;
  margin: 0rem;
  visibility: hidden;
}

.emotion-2 {
  color: hsla(48,7%,86%,1);
  margin-bottom: calc(2rem / 2);
  font-size: 1.2rem;
  font-family: "Open Sans",sans-serif;
  font-weight: 400;
}

.emotion-3 {
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  font-weight: 300;
  line-height: 1.6rem;
  color: hsla(48,7%,86%,1);
}

<a
  class="emotion-8"
  data-test-hasfocus="false"
  data-testid="gallery__thumb"
  href="https://www.google.com"
>
  <div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <div
        class="emotion-5"
      >
        <div
          class="emotion-1"
        >
          <picture>
            <source
              data-testid="picture__source"
              media="(min-width: 300)"
              srcset="./image_3.jpg"
            />
            <source
              data-testid="picture__source"
              media="(min-width: 200)"
              srcset="./image_2.jpg"
            />
            <source
              data-testid="picture__source"
              media="(min-width: 100)"
              srcset="./image_1.jpg"
            />
            <img
              alt="A thumbnail - A brief description."
              class="emotion-0"
              src="./bogus.jpg"
            />
          </picture>
        </div>
        <div
          class="emotion-4"
        >
          <h3
            class="emotion-2"
          >
            A thumbnail
          </h3>
          <p
            class="emotion-3"
          >
            A brief description.
          </p>
        </div>
      </div>
    </div>
  </div>
</a>
`;
